File: Benchmarks/enterprise/python/flask/testcode/benchmark_test_54263.py
# SPDX-License-Identifier: Apache-2.0
from flask import request, jsonify


def BenchmarkTest54263():
    referer_value = request.headers.get('Referer', '')
    data = f'{referer_value:.200s}'
    match str(data):
        case 'a': action = 'alpha'
        case 'b': action = 'beta'
    return jsonify({'action': action}), 200

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest54263.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.io.*;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest54263 {

    @PostMapping(path="/BenchmarkTest54263", consumes="text/plain")
    public void BenchmarkTest54263(@RequestBody String rawBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String rawData = rawBody != null ? rawBody : "";
        java.util.List<String> tokens = java.util.Arrays.asList(rawData.split(","));
        String data = String.join(",", tokens);
        try (java.io.FileWriter fw = new java.io.FileWriter("/var/data/secrets.txt")) {
            fw.write(data);
        }
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}
